mutation_writer/multishard_writer.cc

```diff
@@ -173,6 +173,13 @@ future<> multishard_writer::distribute_mutation_fragments() {
                 return handle_end_of_stream();
             }
         });
+    }).handle_exception([this] (std::exception_ptr ep) {
```

Comment:
Why add more continuations? You can replace `final()` clause in `multishard_writer::operator()` with `then_wrappe()` and handle the error there, no?

Reply:
It is the job of multishard_writer::distribute_mutation_fragments to abort queue. Not the job of the caller.  Also multishard_writer::distribute_mutation_fragments is called once. I do not see a problem attaching a .handle_exception to it.  I also think handle_exception is cleaner than then_wrapper.

Reply:
It may be conceptually cleaner, but it requires more allocations. There is nothing wrong with using then_wrapped to handle both happy and unhappy path if both has to be handled. In fact this is preferred practice. As of hwos job it is - this is all internal multishard_writer code, so callers and callees are  very intervened here. I do have a point that this is not a fast path, so may be extra allocation is not that matter.

But at least make sure that the error is logged. Looking at the code I actually think it will not be.

Reply:
The error is now logged in the second version.